enterprise/server/clientidentity/clientidentity.go

```diff
@@ -91,7 +92,7 @@ func (s *Service) AddIdentityToContext(ctx context.Context) (context.Context, er
 
 func (s *Service) ValidateIncomingIdentity(ctx context.Context) (context.Context, error) {
 	vals := metadata.ValueFromIncomingContext(ctx, authutil.ClientIdentityHeaderName)
-	if len(vals) == 0 {
+	if len(vals) == 0 && !*required {
```

Comment:
I think this should go after validation...

Reply:
Both places actually. May want to extract this into a separate function.

Reply:
whoops sorry i added a test and updated the cl